miio/gateway.py

```diff
@@ -529,45 +581,295 @@ def set_light(self, color_name, brightness):
+class SubDevice(Device):
+    """
+    Base class for all subdevices of the gateway
+    these devices are connected through zigbee.
+    """
 
-    def unpair(self):
-        return self.gw.send("remove_device", [self.sid])
+    def __init__(
+        self,
+        gw: Gateway = None,
+        dev_info: SubDeviceInfo = None,
+        ip: str = None,
+        token: str = None,
+        start_id: int = 0,
+        debug: int = 0,
+        lazy_discover: bool = True,
+    ) -> None:
+        if gw is not None:
+            self._gw = gw
+        else:
+            self._gw = Device(ip, token, start_id, debug, lazy_discover)
+            _LOGGER.debug(
+                "Creating new device instance, only use this for cli interface"
+            )
 
-    def battery(self):
-        return self.gw.send("get_battery", [self.sid])[0]
+        if dev_info is None:
+            raise Exception("sid of the subdevice needs to be specified")
 
-    def get_firmware_version(self) -> Optional[int]:
-        """Returns firmware version"""
+        self._gw = gw
+        self.sid = dev_info.sid
+        self._battery = None
         try:
-            return self.gw.send("get_device_prop", [self.sid, "fw_ver"])[0]
-        except Exception as ex:
-            _LOGGER.debug(
-                "Got an exception while fetching fw_ver: %s", ex, exc_info=True
-            )
-            return None
+            self.type = DeviceType(dev_info.type_id)
+        except ValueError:
+            self.type = DeviceType(-1)
 
     def __repr__(self):
         return "<Subdevice %s: %s fw: %s bat: %s>" % (
-            self.type,
+            self.device_type,
             self.sid,
             self.get_firmware_version(),
-            self.battery(),
+            self.get_battery(),
         )
 
+    @property
+    def device_type(self):
+        """Return the device type name."""
+        return self.type.name
+
+    @property
+    def battery(self):
+        """Return the battery level."""
+        return self._battery
+
+    @command()
+    def subdevice_send(self, command):
+        """Send a command/query to the subdevice"""
+        try:
+            return self._gw.send(command, [self.sid])
+        except Exception as ex:
+            raise GatewayException(
+                "Got an exception while sending command %s" % (command)
+            ) from ex
+            return None
+
+    @command()
+    def subdevice_send_arg(self, command, arguments):
+        """Send a command/query including arguments to the subdevice"""
+        try:
+            return self._gw.send(command, arguments, extra_parameters={"sid": self.sid})
+        except Exception as ex:
+            raise GatewayException(
+                "Got an exception while sending "
+                "command '%s' with arguments '%s'" % (command, str(arguments))
+            ) from ex
+
+    @command(click.argument("property"))
+    def get_subdevice_prop(self, property):
```

Comment:
`get_prop` and `get_prop_*` just like with the main device to be consistent all around?

Reply:
I made it more consistant